ash/src/extensions/khr/pipeline_binary.rs

```diff
@@ -0,0 +1,93 @@
+//! <https://registry.khronos.org/vulkan/specs/1.3-extensions/man/html/VK_KHR_pipeline_binary.html>
+
+use crate::prelude::*;
+use crate::vk;
+use crate::RawPtr as _;
+use alloc::vec::Vec;
+use core::mem;
+
+impl crate::khr::pipeline_binary::Device {
+    /// <https://registry.khronos.org/vulkan/specs/1.3-extensions/man/html/vkCreatePipelineBinariesKHR.html>
+    #[inline]
+    #[doc(alias = "vkCreatePipelineBinariesKHR")]
+    pub unsafe fn create_pipeline_binaries(
+        &self,
+        create_info: &vk::PipelineBinaryCreateInfoKHR<'_>,
+        allocation_callbacks: Option<&vk::AllocationCallbacks<'_>>,
+        binaries: &mut vk::PipelineBinaryHandlesInfoKHR<'_>,
+    ) -> VkResult<()> {
+        (self.fp.create_pipeline_binaries_khr)(
+            self.handle,
+            create_info,
+            allocation_callbacks.as_raw_ptr(),
+            binaries,
+        )
+        .result()
+    }
+
+    /// <https://registry.khronos.org/vulkan/specs/1.3-extensions/man/html/vkDestroyPipelineBinaryKHR.html>
+    #[inline]
+    #[doc(alias = "vkDestroyPipelineBinaryKHR")]
+    pub unsafe fn destroy_pipeline_binary(
+        &self,
+        pipeline_binary: vk::PipelineBinaryKHR,
+        allocation_callbacks: Option<&vk::AllocationCallbacks<'_>>,
+    ) {
+        (self.fp.destroy_pipeline_binary_khr)(
+            self.handle,
+            pipeline_binary,
+            allocation_callbacks.as_raw_ptr(),
+        )
+    }
+
+    /// <https://registry.khronos.org/vulkan/specs/1.3-extensions/man/html/vkGetPipelineKeyKHR.html>
+    #[inline]
+    #[doc(alias = "vkGetPipelineKeyKHR")]
+    pub unsafe fn get_pipeline_key(
+        &self,
+        pipeline_create_info: Option<&vk::PipelineCreateInfoKHR<'_>>,
+    ) -> VkResult<vk::PipelineBinaryKeyKHR<'_>> {
```

Comment:
Shouldn't `vk::PipelineBinaryKeyKHR` be an out parameter, to allow for extension? If this is for some reason undesired, then the lifetime should be `'static`.